cmd/mimir/main.go

```diff
@@ -319,6 +319,11 @@ func expandEnvironmentVariables(config []byte) []byte {
 		if v == "" && len(keyAndDefault) == 2 {
 			v = keyAndDefault[1] // Set value to the default.
 		}
+
+		if strings.Contains(v, "\n") {
```

Comment:
nit: Could this just be `return strings.Replace(v, "\n", "", -1)`, without the conditional?

Reply:
I wanted to do that only to apply this transformation to files with `\n`, so we reduce the risk of regression until a potentially better solution is found. 

Because it's just a nit, I'm opening the discussion, but if you want me to apply this modification, it's not a problem.